src/server/rtu.rs

```diff
@@ -90,13 +94,20 @@ where
         }?;
 
         let hdr = request.hdr;
-        let response = service.call(request.pdu.0).await.map_err(Into::into)?;
-        framed
-            .send(rtu::ResponseAdu {
-                hdr,
-                pdu: response.into(),
-            })
-            .await?;
+        let response: OptionalResponsePdu = service
+            .call(request.into())
+            .await
+            .map_err(Into::into)?
+            .into();
+
+        match response.0 {
+            Some(pdu) => {
+                framed.send(rtu::ResponseAdu { hdr, pdu }).await?;
+            }
+            None => {
+                log::debug!("skipping response");
```

Comment:
This log message could be rewritten to better reflect the context and outside view instead of reflecting implementation details. How about `"No response for request {hdr:?}"`

Personal preference: I tend to start log messages with a capital letter, like a Git commit message.

Reply:
yes, logging the hdr is a good idea, I wanted to log the error from TryInto but I gave up and forgot :sweat_smile: